File: botNET.0.5.sources/botnet/botNET/Distance.cpp
#include "botnet.h"
#include ".\distance.h"
#include "distance.h"
#include "unitbase.h"




namespace botNET
{
	namespace Mapping
	{

		float Distance::Manhattan(float x1, float y1, float x2, float y2) 
		{
			return Math::Abs(x2-x1) + Math::Abs(y2-y1);
		}

		int Distance::Manhattan(int x1, int y1, int x2, int y2) 
		{
			return Math::Abs(x2-x1) + Math::Abs(y2-y1);
		}

		float Distance::Manhattan(System::Drawing::Point c1, System::Drawing::Point c2) 
		{
			return Manhattan((float)c1.X, (float)c1.Y, (float)c2.X, (float)c2.Y);
		}

		float Distance::Euclidian(float x1, float y1, float x2, float y2) 
		{
			return (float)Math::Sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
		}

		int Distance::Euclidian(int x1, int y1, int x2, int y2) 
		{
			return (int)Math::Sqrt((double)((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2)));
		}

		float Distance::Euclidian(System::Drawing::Point c1, System::Drawing::Point c2) 
		{
			return Manhattan((float)c1.X, (float)c1.Y, (float)c2.X, (float)c2.Y);
		}

		float Distance::EuclidianNoSqrt(float x1, float y1, float x2, float y2) 
		{
			return (float)Math::Floor((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
		}

		int Distance::EuclidianNoSqrt(int x1, int y1, int x2, int y2) 
		{
			return (int)Math::Floor((double)((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2)));
		}

		float Distance::EuclidianNoSqrt(System::Drawing::Point c1, System::Drawing::Point c2) 
		{
			return EuclidianNoSqrt((float)c1.X, (float)c1.Y, (float)c2.X, (float)c2.Y);
		}

		float Distance::DiagShortcut(float x1, float y1, float x2, float y2) 
		{
			float dy = Math::Abs(y2-y1);
			float dx = Math::Abs(x2-x1);
			if (dy <= dx) return (float)(dy*0.414 + dx); else return (float)(dx*0.414 + dy);
		}

		int Distance::DiagShortcut(int x1, int y1, int x2, int y2) 
		{
			float dy = (float)Math::Abs(y2-y1);
			float dx = (float)Math::Abs(x2-x1);
			if (dy <= dx) return (int)(dy*0.414 + dx); else return (int)(dx*0.414 + dy);
		}

		float Distance::DiagShortcut(System::Drawing::Point c1, System::Drawing::Point c2) 
		{
			return DiagShortcut((float)c1.X, (float)c1.Y, (float)c2.X, (float)c2.Y);
		}

		float Distance::MaxDxDy(float x1, float y1, float x2, float y2) 
		{
			float dy = Math::Abs(y2-y1);
			float dx = Math::Abs(x2-x1);
			return (dx>dy)?dx:dy;
		}

		int Distance::MaxDxDy(int x1, int y1, int x2, int y2) 
		{
			int dy = Math::Abs(y2-y1);
			int dx = Math::Abs(x2-x1);
			return (dx>dy)?dx:dy;
		}

		float Distance::MaxDxDy(System::Drawing::Point c1, System::Drawing::Point c2) 
		{
			return MaxDxDy((float)c1.X, (float)c1.Y, (float)c2.X, (float)c2.Y);
		}

		float Distance::GetAngle(float x1, float y1, float x2, float y2) 
		{
			return (float)Math::Floor((Math::Atan2(y2-y1 , x2-x1)) * 180 / 3.1415);
		}

		float Distance::GetAngle(System::Drawing::Point c1, System::Drawing::Point c2) 
		{
			return (float)Math::Floor((Math::Atan2((float)c2.Y-(float)c1.Y , (float)c2.X-(float)c1.X)) * 180 / 3.1415);
		}

	}

}

File: botNET.Pathing.2.0.3/Distance.cs
using System;

using botNET.Mapping;

namespace botNET.Pathing
{
	/// <summary>
	/// Summary description for Distance.
	/// </summary>
	public class Distance
	{

		/// <summary>
		/// Returns the Distance using the Manhattan Method
		/// </summary>
		/// <param name="x1"></param>
		/// <param name="y1"></param>
		/// <param name="x2"></param>
		/// <param name="y2"></param>
		/// <returns></returns>
		public static int Manhattan(int x1, int y1, int x2, int y2) 
		{
			return Math.Abs(x2-x1) + Math.Abs(y2-y1);
		}
		/// <summary>
		/// Returns the Distance using the Manhattan Method
		/// </summary>
		/// <param name="c1"></param>
		/// <param name="c2"></param>
		/// <returns></returns>
		public static int Manhattan(System.Drawing.Point c1, System.Drawing.Point c2) 
		{
			return Manhattan(c1.X, c1.Y, c2.X, c2.Y);
		}


		/// <summary>
		/// Returns the Distance using the Euclidian Method
		/// </summary>
		/// <param name="x1"></param>
		/// <param name="y1"></param>
		/// <param name="x2"></param>
		/// <param name="y2"></param>
		/// <returns></returns>
		public static int Euclidian(int x1, int y1, int x2, int y2) 
		{
			return (int)Math.Sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
		}

		/// <summary>
		/// Returns the Distance using the Euclidian Method
		/// </summary>
		/// <param name="c1"></param>
		/// <param name="c2"></param>
		/// <returns></returns>
		public static int Euclidian(System.Drawing.Point c1, System.Drawing.Point c2) 
		{
			return Manhattan(c1.X, c1.Y, c2.X, c2.Y);
		}


		/// <summary>
		/// Returns the Distance using the Euclidian Method with no Square Root
		/// </summary>
		/// <param name="x1"></param>
		/// <param name="y1"></param>
		/// <param name="x2"></param>
		/// <param name="y2"></param>
		/// <returns></returns>
		public static int EuclidianNoSqrt(int x1, int y1, int x2, int y2) 
		{
			return (int)Math.Floor((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2));
		}

		/// <summary>
		/// Returns the Distance using the Euclidian Method with no Square Root
		/// </summary>
		/// <param name="c1"></param>
		/// <param name="c2"></param>
		/// <returns></returns>
		public static int EuclidianNoSqrt(System.Drawing.Point c1, System.Drawing.Point c2) 
		{
			return EuclidianNoSqrt(c1.X, c1.Y, c2.X, c2.Y);
		}


		/// <summary>
		/// Returns the Distance using the Diagonal Shortcut Method. This offers best Measurement for Diablo Type games
		/// </summary>
		/// <param name="x1"></param>
		/// <param name="y1"></param>
		/// <param name="x2"></param>
		/// <param name="y2"></param>
		/// <returns></returns>
		public static int DiagShortcut(int x1, int y1, int x2, int y2) 
		{
			int dy = Math.Abs(y2-y1);
			int dx = Math.Abs(x2-x1);
			if (dy <= dx) return (int)(dy*0.414 + dx); else return (int)(dx*0.414 + dy);
		}

		/// <summary>
		/// Returns the Distance using the Diagonal Shortcut Method. This offers best Measurement for Diablo Type games
		/// </summary>
		/// <param name="c1"></param>
		/// <param name="c2"></param>
		/// <returns></returns>
		public static int DiagShortcut(System.Drawing.Point c1, System.Drawing.Point c2) 
		{
			return DiagShortcut(c1.X, c1.Y, c2.X, c2.Y);
		}


		/// <summary>
		/// Returns the Distance using the Maximum DxDy Method.
		/// </summary>
		/// <param name="x1"></param>
		/// <param name="y1"></param>
		/// <param name="x2"></param>
		/// <param name="y2"></param>
		/// <returns></returns>
		public static int MaxDxDy(int x1, int y1, int x2, int y2) 
		{
			int dy = Math.Abs(y2-y1);
			int dx = Math.Abs(x2-x1);
			return (dx>dy)?dx:dy;
		}

		/// <summary>
		/// Returns the Distance using the Maximum DxDy Method.
		/// </summary>
		/// <param name="c1"></param>
		/// <param name="c2"></param>
		/// <returns></returns>
		public static int MaxDxDy(System.Drawing.Point c1, System.Drawing.Point c2) 
		{
			return MaxDxDy(c1.X, c1.Y, c2.X, c2.Y);
		}

		/// <summary>
		/// Returns an angle
		/// </summary>
		/// <param name="x1"></param>
		/// <param name="y1"></param>
		/// <param name="x2"></param>
		/// <param name="y2"></param>
		/// <returns></returns>
		public static double GetAngle(double x1, double y1, double x2, double y2) 
		{
			return Math.Floor((Math.Atan2(y2-y1 , x2-x1)) * 180 / 3.1415);
		}

	}
}
